src/main/java/org/opensearch/knn/index/KNNWeight.java

```diff
@@ -110,7 +110,8 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
             } else {
                 String engineName = fieldInfo.attributes().getOrDefault(KNN_ENGINE, KNNEngine.NMSLIB.getName());
                 knnEngine = KNNEngine.getEngine(engineName);
-                spaceType = SpaceType.getSpace(fieldInfo.getAttribute(SPACE_TYPE));
+                String spaceTypeName = fieldInfo.attributes().getOrDefault(SPACE_TYPE, SpaceType.L2.getValue());
```

Comment:
```suggestion
                String spaceTypeName = fieldInfo.attributes().getOrDefault(SPACE_TYPE, SpaceType.DEFAULT.getValue());
```

Reply:
I actually left L2 intentionally. This comes up if segment create in 7.1, or 7.4. At that time, L2 was the only space available. If I set it to DEFAULT and for whatever reason default changes, this could get messed up.

Reply:
got it. If i remember correctly, the reason we kept default parameter is only for backward compatibility right?

Reply:
Understood, but I prefer to be explicit in this case. 

I dont think its a major concern, but I see DEFAULT as fall back to this value if the user did not provide a value in some input. However, in this case, the only reason a value would not be here is if the segments were created in 7.1 or 7.4. In those cases, we have to assume the user intended to use L2